### ISSUE TITLE
Plugin chaining

### ISSUE THREAD

**Issue by AndersDJohnson** [jnordberg/wintersmith#144]
Is there a way to compose multiple content plugins, e.g. pass some resource through multiple plugin handlers, or is it essentialy first match wins based on the glob patterns from `registerContentPlugin()`?

**Comment by dtinth** [comment]
I came across this issue when I want to use Stylus with Autoprefixer. I’ve solved this by creating a gulp plugin adaptor. I called it [vinylsmith].

With [vinylsmith], you can create Wintersmith plugins that pipes files through a series of Gulp plugins very easily like this:

```coffee
vinylsmith = require 'vinylsmith'

stylus   = require 'gulp-stylus'
pleeease = require 'gulp-pleeease'
rev      = require 'rev'

module.exports = (env, callback) ->

  env.registerContentPlugin 'styles', '**/*.styl',
    vinylsmith(env)
      .pipe(stylus, 'include css': true)   # compile stylus
      .pipe(pleeease)                      # autoprefix, minify
      .pipe(rev)                           # asset-hashing

  callback()
```

Now your `main.styl` will get [compiled][gulp-stylus] into `main-1682966244.css`, postprocessed and minified with [pleeease], and filename hashed with [gulp-rev].

[pleeease]: http://pleeease.io/
[gulp-stylus]: https://www.npmjs.com/package/gulp-stylus
[gulp-rev]: https://www.npmjs.com/package/gulp-rev
[vinylsmith]: https://github.com/dtinth/vinylsmith

**Comment by fidian** [comment]
Composing a plugin from others seems like a good solution for a one-off project.  The drawback I see is that I'd have to make a new Wintersmith plugin for every combination of things that I'd like.  If I have two different projects and they have different requirements, I'd have to make two different plugins.  The plugins I generate wouldn't be as useful for friends because they may want an extra processing step or for the order/options to be changed.

**Comment by dtinth** [comment]
That’s true. But just as different project uses different `gulpfile.js`, so I think it is the same with Wintersmith. Different static sites have different requirements for processing pipeline.

If creating such plugin is painless, then I think it wouldn’t be a problem if each static site has their own plugin. In fact, this is what I am doing with vinylsmith: to make creating Wintersmith plugin as trivial as writing a gulpfile, so that each static site can have its own plugin.

**Comment by fkrauthan** [comment]
Why don't implement such a plugin grouping/pipeline within wintersmith? Creating a way to chain multiple plugins and export that as an "virtual" plugin that can be applied to pages?
